app/assets/javascripts/govuk_publishing_components/components/details.js

```diff
@@ -1,16 +1,19 @@
 /* eslint-env jquery */
 // = require govuk/components/details/details.js
 window.GOVUK = window.GOVUK || {}
-window.GOVUK.Modules = window.GOVUK.Modules || {};
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+window.GOVUK.Modules.Details = window.GOVUKFrontend;
```

Comment:
Good find! It seems like this was [missed when the details script was brought over from `govuk-frontend`](https://github.com/alphagov/govuk_publishing_components/commit/6112261619d47d8f930e58387b66023deb657cc7#diff-a235b89aac8e8310ee228b267a5837df7801e2c6510b4f9772077a88abbf56c0R5). This line should be enough to fix the problem; the initialisation will be automatically done via [`modules.js`](https://github.com/alphagov/govuk_publishing_components/blob/master/app/assets/javascripts/govuk_publishing_components/modules.js#L48).

Reply:
> This line should be enough to fix the problem

@alex-ju I would have thought so too, in fact that was the first thing I tried, however it didn't seem to be enough. I've done  some digging and I think what is happening here is:
- the details component has `data-module="govuk-details"`, therefore in the [modules.js start function](https://github.com/alphagov/govuk_publishing_components/blob/176d844f0085ccb8bcf6bc808987e26cdcd17530/app/assets/javascripts/govuk_publishing_components/modules.js#L31), `moduleName` is set as `GovukDetails` and `frontendModuleName` is `Details`
- the [first condition](https://github.com/alphagov/govuk_publishing_components/blob/176d844f0085ccb8bcf6bc808987e26cdcd17530/app/assets/javascripts/govuk_publishing_components/modules.js#L35) in the if statement evaluates to `true` because `GOVUK.Modules[GovukDetails]` exists and it has an `init` function –  so the code at L40-L42 runs
- since the condition before it was true, the `else if` below does not run. Therefore the GOVUK Frontend `Details` module does not get initialised even though it exists and is "initialisable" 😄

Reply:
Basically I think there’s a bigger issue with how we init modules. What I did is really a workaround. Not sure if we should try and go with the workaround for now and try to resolve the core issue later. Let me know what you think, anyway.

Reply:
Oh, that is bad! I was wondering how other scripts work (for example checkboxes) and I was surprised to learn that we use [an extra `data-module` attribute](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_checkboxes.html.erb#L26) at a lower level in the markup tree 🙈. Yes please, splitting the conditional statement in `modules.js` sounds like a good idea.

Reply:
If you have other things you're currently working on let me know and I can pick this up. However works best for you.

Reply:
Thanks for offering @alex-ju, I've just pushed the latest changes to this branch. 
Will be updating the PR desc to reflect what's changed.